api/scholarqa/rag/retrieval.py

```diff
@@ -40,7 +42,17 @@ def retrieve_passages(self, query: str, **filter_kwargs) -> List[Dict[str, Any]]
     def retrieve_additional_papers(self, query: str, **filter_kwargs) -> List[Dict[str, Any]]:
         if self.max_date:
             # S2 API doesnt have insertedBefore for this endpoint, so use publicationDateOrYear:
-            filter_kwargs.update({"publicationDateOrYear": ':{}'.format(self.max_date)})
+            if year_filter := filter_kwargs.pop("year", None):
+                # if year filter is present, we need to remove it from the kwargs and resolve it to publicationDateOrYear
+                date_start, date_end = year_filter.split("-") # YYYY-YYYY
+                max_year = self.max_date.split("-")[0] #YYYY-MM
+                # restrict both start and end to max_year and if end is max_year, set it to max_date
+                date_start, date_end = min(date_start, max_year), date_end if date_end and date_end < max_year else self.max_date
```

Comment:
One more question about desired behavior here... Suppose we have something where the desired date range is completely after the desired inserted before date... E.g. inserted before/max_date = 2025-05, year range = 2026-2028... What do we want to happen there? As far as I can tell we'll get something like 2025:2025-05 (so basically anything in 2025 through May). Is that what we want?